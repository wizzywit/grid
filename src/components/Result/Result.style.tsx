import styled from "styled-components";

export const Styles = {
  ResultContainer: styled.div`
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    /* justify-content: space-between;
    width: 50%; */
    gap: 15px;
    margin-bottom: 50px;
  `,
};
