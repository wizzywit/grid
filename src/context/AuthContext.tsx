import React from "react";

interface DataArray {
  color: string;
  shape: string;
}

export type Filter = (
  c: Array<string> | ((c: Array<string>) => Array<string>)
) => void;

type PerformFilterType = (
  shape: string,
  shapesFilter: Array<string> | undefined,
  filter: Filter | undefined,
  type: string
) => void;

type AuthContextType = {
  login?: () => void;
  logout?: () => void;
  data?: Array<DataArray>;
  shapesFilter?: Array<string>;
  setShapesFilter?: Filter;
  colorsFilter?: Array<string>;
  setColorsFilter?: Filter;
  isLoggedIn?: boolean;
  performSelect?: PerformFilterType;
};
const AuthContext = React.createContext<AuthContextType>({});

export default AuthContext;
