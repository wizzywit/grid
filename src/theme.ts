import { DefaultTheme } from "styled-components";

const myTheme: DefaultTheme = {
  borderRadius: "5px",

  colors: {
    main: "#f5f6fc",
    secondary: "white",
    title: "#5b95f7",
  },
};

export { myTheme };
