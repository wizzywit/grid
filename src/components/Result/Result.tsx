import React, { useContext } from "react";
import { Styles } from "./Result.style";
import Item from "../Item/Item";
import AuthContext from "../../context/AuthContext";

const Result = () => {
  const { data } = useContext(AuthContext);
  return (
    <Styles.ResultContainer>
      {data?.map((datum, i) => (
        <Item key={i} shape={datum.shape} color={datum.color} />
      ))}
    </Styles.ResultContainer>
  );
};

export default Result;
