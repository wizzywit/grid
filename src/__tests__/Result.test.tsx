import React from "react";

import { render } from "@testing-library/react";

import Result from "../components/Result/Result";
import AuthContext from "../context/AuthContext";
import { ThemeProvider } from "styled-components";
import { myTheme } from "../theme";

const data = [
  { shape: "oval", color: "red" },
  { shape: "traingle", color: "grey" },
  { shape: "round", color: "blue" },
  { shape: "square", color: "green" },
  { shape: "rectangle", color: "grey" },
];

describe("Result", () => {
  test("should display list of items", async () => {
    const { container, findAllByTestId } = render(
      <AuthContext.Provider value={{ data }}>
        <ThemeProvider theme={myTheme}>
          <Result />
        </ThemeProvider>
      </AuthContext.Provider>
    );
    expect(container).toMatchSnapshot();
    const children = await findAllByTestId("item");
    expect(children).toHaveLength(data.length);
  });
});
