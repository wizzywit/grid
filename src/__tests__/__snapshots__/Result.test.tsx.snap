// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should display list of items 1`] = `
<div>
  <div
    class="sc-bdnxRM kiMljk"
  >
    <div
      class="sc-gtsrHT kqzUsG"
      color="red"
      data-testid="item"
    >
      <div
        class="oval"
        data-shape="oval"
        data-testid="shape"
      />
    </div>
    <div
      class="sc-gtsrHT eOLTIH"
      color="grey"
      data-testid="item"
    >
      <div
        class="traingle"
        data-shape="traingle"
        data-testid="shape"
      />
    </div>
    <div
      class="sc-gtsrHT jZZkxs"
      color="blue"
      data-testid="item"
    >
      <div
        class="round"
        data-shape="round"
        data-testid="shape"
      />
    </div>
    <div
      class="sc-gtsrHT bgdPXS"
      color="green"
      data-testid="item"
    >
      <div
        class="square"
        data-shape="square"
        data-testid="shape"
      />
    </div>
    <div
      class="sc-gtsrHT eOLTIH"
      color="grey"
      data-testid="item"
    >
      <div
        class="rectangle"
        data-shape="rectangle"
        data-testid="shape"
      />
    </div>
  </div>
</div>
`;
